### PR TITLE
Fix up surface area tests

### PR DESCRIPTION
**Pull request by latkin**
Fixes up and improves the FSharp.Core surface area tests:

- All portable profiles now covered
- Reduced code duplication
- Better logging when something needs to be updated
  - Suggests `windiff` command line to run to see delta between actual and expected

### CODE REVIEW

**Comment by enricosada** [comment]
you can cleanup using [nunit testcaseattribute](http://www.nunit.org/index.php?p=testCaseSource&r=2.5) using only one test (like yours but without an additional module)

```
module SurfaceAreaTest =

    let profiles = 
         [  "portable47", SurfaceArea47.area;
             "net40", SurfaceArea40.area;
             "net20", SurfaceArea20.area;
         ]
         |> List.map (fun (p,e) -> TestCaseData(p, e))
         |> Seq.ofList

    [<Test; TestCaseSource("profiles")>]
    let VerifyArea profile logFilePath expected =
         //inline the test code here
         SurfaceArea.verify expected profile logFilePath 
```

File: src/fsharp/FSharp.Core.Unittests/SurfaceArea.net20.fs [hunk 1]
```diff
@@ -3118,8 +3278,4 @@ System.Tuple`8[T1,T2,T3,T4,T5,T6,T7,TRest]: TRest Rest
 System.Tuple`8[T1,T2,T3,T4,T5,T6,T7,TRest]: TRest get_Rest()
 System.Tuple`8[T1,T2,T3,T4,T5,T6,T7,TRest]: Void .ctor(T1, T2, T3, T4, T5, T6, T7, TRest)
 "
-
-            let act = actual.ToString()
-            if expected <> act then
-                printf "%s" act
-            Assert.AreEqual(expected, act)
+        SurfaceArea.verify expected "net20" (sprintf "%s\\%s" __SOURCE_DIRECTORY__ __SOURCE_FILE__)
```
Comment (enricosada) [inline src/fsharp/FSharp.Core.Unittests/SurfaceArea.net20.fs]: missing newline at EOF

File: src/fsharp/FSharp.Core.Unittests/SurfaceArea.portable47.fs [hunk 2]
```diff
@@ -3430,26 +3428,6 @@ Microsoft.FSharp.Reflection.UnionCaseInfo: System.Type get_DeclaringType()
...
-            printfn "Next bit is"
-            printfn "Exp: %s" (expected.Substring(indexFirstDiff, 400).Replace("\n","     "))
-            printfn "Act: %s" (act.Substring(indexFirstDiff, 400).Replace("\n","     "))
-            printfn "Full actual below"
-            printfn ""
-            printf "%s" act
-        Assert.AreEqual(expected, act)
+System.IObserver`1[T]: Void OnNext(T)
+"
+        SurfaceArea.verify expected "portable47" (sprintf "%s\\%s" __SOURCE_DIRECTORY__ __SOURCE_FILE__)
```
Comment (enricosada) [inline src/fsharp/FSharp.Core.Unittests/SurfaceArea.portable47.fs]: missing newline at EOF

Comment (enricosada) [inline src/fsharp/FSharp.Core.Unittests/SurfaceArea.portable47.fs]: missing newline at EOF

**Comment by enricosada** [comment]
anyway, bugfix and refactoring shouldn't be different pull request? easier to diff and review

**Comment by latkin** [comment]
@dsyme I am hitting some kind of newline inconsistency thing in AppVeyor, did you have a solution for that?

```
4313  Expected string length 372441 but was 375884. Strings differ at index 73.
4314  Expected: "... Boolean Equals(System.Object)\nMicrosoft.FSharp.Collection..."
4315  But was:  "... Boolean Equals(System.Object)\r\nMicrosoft.FSharp.Collectio..."
4316  ---------------------------------------------^
```

**Comment by forki** [comment]
in Paket, FAKE,.. and a lot of other projects we normalize the generated strings since it wil be always different on travis vs. appveyor.

**Comment by latkin** [comment]
Mmm, yes probably easiest/most reliable to just update the `normalize` guy to fix up newlines no matter what.  Will do...

**Comment by latkin** [comment]
@enricosada Our internal build/test still relies on nunit 2.2.0, so this dark magic is not available for us :fearful:

**Comment by latkin** [comment]
Ok I think newline stuff should be fixed now, will see how this build goes.

@enricosada I actually prefer *not* appending unnecessary newlines, but I added them just for you anyway :heart:

**Comment by enricosada** [comment]
ty @latkin , i get problem/noise without newlines ( with cat, git diff complain etc )

**Comment by dsyme** [comment]
@latkin - Yes, I hit that, the solution was https://github.com/Microsoft/visualfsharp/commit/a86d5c46124b1a476e0baefd8d9a4c5a84d603dc#diff-3d8017ffe39ef440a86d4f58517a82e0L3482

You may need the same thing in SurfaceArea.2.0.fs

I don't know what caused it. Perhaps AppVeyor always checks out with LF or CRLF on Windows,  thinking it's doing Windows devs a favour.

**Pull request closed without merging**